hazelcast/src/main/java/com/hazelcast/sql/impl/expression/Expression.java

```diff
@@ -20,12 +20,15 @@
 import com.hazelcast.sql.impl.row.Row;
 import com.hazelcast.sql.impl.type.QueryDataType;
 
+import java.io.Serializable;
+
 /**
  * Defines expression contract for SQL.
  *
  * @param <T> the return type of this expression.
  */
-public interface Expression<T> extends DataSerializable {
+public interface Expression<T> extends DataSerializable, Serializable {
```

Comment:
Does Jet still require `Expression` to be serializable? Ideally, we should avoid adding `Serializable` here, since this is just an internal class that is passed over the wire, and is not meant to be serialized by some generic mechanism in IMDG

Reply:
I tried to fix this. It's doable, but it's extremely lot of extra code and the gain is little. We have to replace a simple serializable lambda with a class that implements `DataSerializable` and writes all the captured state manually. For example:
![image](https://user-images.githubusercontent.com/9109877/90526125-4ba65480-e170-11ea-85fa-0e7ac988f9b1.png)
I started with it but stopped after few hours with the light at the end of the tunnel still far ahead. We use lambdas everywhere and often wrap them in another lambdas to modify some behavior.

The benefit is little: it makes the job initialization slightly faster, but Jet never aimed at maximum job deployment performance. My guess is that the speedup is less than 1ms per job.

Reply:
Actually, I removed it from `QueryPath`